kairos-tx/src/lib.rs

```diff
@@ -4,3 +4,28 @@ extern crate alloc;
 pub mod asn;
 pub mod error;
 pub mod helpers;
+
+use sha2::Digest;
+
+// Computes the hash for a transaction.
+// Hash is obtained from payload by computing sha256 of DER encoded ASN.1 data.
+pub fn hash(payload: &asn::SigningPayload) -> Result<[u8; 32], error::TxError> {
+    let data = rasn::der::encode(payload).map_err(error::TxError::EncodeError)?;
+    let tx_hash: [u8; 32] = sha2::Sha256::digest(data).into();
+    Ok(tx_hash)
+}
+
+// Constructor for non-exhaustive transaction struct.
+pub fn make_tx(
```

Comment:
I don't understand the purpose of this function. it is basically `new` for a `Transaction` type. Not saying it should be a `new`.

Reply:
You are right. I replaced this helper with constructor in https://github.com/cspr-rad/kairos/pull/74/commits/52b2854fb6949adc3d5009e6bcca7a37578cec4c.